DigitalLearningSolutions.Data.Tests/DataServices/RegistrationDataServiceTests.cs

```diff
@@ -209,6 +209,89 @@ void Action()
+            const int userId = 2;
+
+            // When
+            var (delegateId, candidateNumber) =
+                service.RegisterDelegateAccountAndCentreDetailForExistingUser(
+                    delegateRegistrationModel,
+                    userId,
+                    currentTime
+                );
+            var user = await connection.GetDelegateUserByCandidateNumberAsync(candidateNumber);
+
+            // Then
+            using (new AssertionScope())
+            {
+                user.Id.Should().Be(delegateId);
+                user.FirstName.Should().Be(delegateRegistrationModel.FirstName);
+                user.LastName.Should().Be(delegateRegistrationModel.LastName);
+                user.EmailAddress.Should().Be(delegateRegistrationModel.PrimaryEmail);
+                user.CentreId.Should().Be(delegateRegistrationModel.Centre);
+                user.Answer1.Should().Be(delegateRegistrationModel.Answer1);
+                user.Answer2.Should().Be(delegateRegistrationModel.Answer2);
+                user.Answer3.Should().Be(delegateRegistrationModel.Answer3);
+                user.Answer4.Should().Be(delegateRegistrationModel.Answer4);
+                user.Answer5.Should().Be(delegateRegistrationModel.Answer5);
+                user.Answer6.Should().Be(delegateRegistrationModel.Answer6);
+                user.DateRegistered.Should().BeCloseTo(currentTime, 100);
+                candidateNumber.Should().Be("FS352");
+                user.CandidateNumber.Should().Be("FS352");
+            }
+        }
+
+        [Test]
+        public async Task
+            RegisterDelegateAccountAndCentreDetailForExistingUser_does_not_create_UserCentreDetails_with_null_centre_specific_email()
+        {
+            using var transactionScope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled);
+
+            // Given
+            var delegateRegistrationModel =
+                RegistrationModelTestHelper.GetDefaultDelegateRegistrationModel(
+                    "forename",
+                    "surname",
+                    "test@gmail.com",
+                    3,
+                    centreSpecificEmail: null
+                );
+            var currentTime = DateTime.Now;
+            const int userId = 2;
+
+            // When
+            var (delegateId, candidateNumber) =
+                service.RegisterDelegateAccountAndCentreDetailForExistingUser(
+                    delegateRegistrationModel,
+                    userId,
+                    currentTime
+                );
+            var userCentreDetailsCount = connection.QuerySingle<int>(
+                "SELECT COUNT(*) FROM UserCentreDetails WHERE CentreID = 3 AND UserID = 2"
+            );
+            var user = await connection.GetDelegateUserByCandidateNumberAsync(candidateNumber);
+
+            // Then
+            user.Id.Should().Be(delegateId);
+            userCentreDetailsCount.Should().Be(0);

```

Comment:
Feels like we should be checking that there's no UserCentreDetails entity with a non-null email because that's the minimum requirement. The app will work perfectly well if the registration does create a UserCentreDetails entity with a null email.